contracts/bonding/BondingManager.sol

```diff
@@ -220,12 +220,18 @@ contract BondingManager is ManagerProxyTarget, IBondingManager {
 
         uint256 currentRound = roundsManager().currentRound();
 
-        if (delegatorStatus(msg.sender) == DelegatorStatus.Unbonded) {
+        if (delegatorStatus(msg.sender) != DelegatorStatus.Bonded) {
             // New delegate
             // Set start round
             del.startRound = currentRound.add(1);
         }
 
+        if (delegatorStatus(msg.sender) == DelegatorStatus.Unbonding) {
+            // If transitioning from unbonding state
+            // make sure to zero out withdraw round
+            del.withdrawRound = 0;
```

Comment:
If a delegator was in the unbonding state and chooses to bond again before the unbonding period is over we reset its withdraw round to 0 - a delegator should only have a withdraw round if it is in the unbonding or unbonded state